newIDE/app/src/Utils/ProjectCache.js

```diff
@@ -32,38 +53,91 @@ class ProjectCache {
+            );
+            reject();
+          };
+        }
+      };
+    });
+  }
+
   _initializeDatabase() {
     if (!this.databasePromise) {
       this.databasePromise = new Promise<IDBDatabase>((resolve, reject) => {
-        const request = window.indexedDB.open(CLOUD_PROJECT_AUTOSAVE_CACHE_KEY);
-        request.onsuccess = event => {
-          if (
-            !event.target.result.objectStoreNames.contains(objectStoreScope)
-          ) {
-            // The onUpgradeNeeded is called before the success event so the object
-            // store should exist.
+        ProjectCache._removeDatabaseIfCorrupt()
+          .then(() => {
+            const request = window.indexedDB.open(
+              CLOUD_PROJECT_AUTOSAVE_CACHE_KEY
+            );
+            request.onsuccess = event => {
+              const db = event.target.result;
+              if (!db.objectStoreNames.contains(objectStoreScope)) {
+                // The onUpgradeNeeded is called before the success event so the object
+                // store should exist.
+                console.error(
+                  `Couldn't find the object store ${objectStoreScope}. An issue must have happened when creating the database.`
+                );
+              }
+              resolve(db);
+            };
+            request.onerror = event => {
+              console.error('IndexedDB could not be opened:', event);
+              reject(event);
+            };
+            request.onupgradeneeded = event => {
+              const db = event.target.result;
+
+              if (!db.objectStoreNames.contains(objectStoreScope)) {
+                db.createObjectStore(objectStoreScope, { keyPath: keyName });
+              }
+            };
+          })
+          .catch(error => {
             console.error(
-              `Couldn't find the object store ${objectStoreScope}. An issue must have happened when creating the database.`
+              'An error occurred while clearing a corrupt database.',
+              error
             );
-          }
-          resolve(event.target.result);
-        };
-        request.onerror = event => {
-          console.error('IndexedDB could not be opened:', event);
-          reject(event);
-        };
-        request.onupgradeneeded = event => {
-          const db = event.target.result;
-
-          if (!db.objectStoreNames.contains(objectStoreScope)) {
-            // The object store can only be created in the onUpgradeNeeded event.
-            // This event is called after the database is created and before the success event.
-            // If, for some reason, the object store creation failed at the time the
-            // database was created, the database will be in a transition state and
-            // ProjectCache instances will always fail.
-            db.createObjectStore(objectStoreScope, { keyPath: keyName });
-          }
-        };
+          });
```

Comment:
this catch is showing a wrong error message. It's either called if `_removeDatabaseIfCorrupt` throws or if the `.then(() => ...` throws.

Instead, use `then(() => ..., error => ...)` so that the second argument is only called if `_removeDatabaseIfCorrupt` throws.
If you want to always catch all errors, update the error message to be generic.

Reply:
Thanks!